Sources/ForageSDK/ForageSDK.swift

```diff
@@ -6,28 +6,42 @@
 import Foundation
 
 public class ForageSDK {
-    // MARK: Properties
+    // MARK: - Static Properties
 
+    // These belong to ForageSDK itself and are shared by all of its instances
     private static var config: Config?
     static var logger: ForageLogger?
-    var service: ForageService?
-    var merchantID: String = config?.merchantID ?? ""
-    var sessionToken: String = config?.sessionToken ?? ""
-    var traceId: String = ""
-
-    public var environment: Environment = .sandbox
-    // Don't update! Only updated when releasing.
+    // Don't update version! Only updated when releasing.
     public static let version = "4.4.9"
     public static let shared = ForageSDK()
+    
+    // MARK: - Singleton Instance Properties
+
+    // Instance properties, only belong to the `shared` singleton instance.
+    var service: ForageService?
+    
+    // Computed instance properties derived from static values
+    public var environment: Environment {
+        Environment(sessionToken: ForageSDK.config?.sessionToken)
+    }
+    var merchantID: String {
+        ForageSDK.config?.merchantID ?? ""
+    }
+    var sessionToken: String {
+        ForageSDK.config?.sessionToken ?? ""
+    }
+    var traceId: String {
+        ForageSDK.logger?.getTraceID() ?? ""
+    }
```

Comment:
These computed properties work by running the closure code whenever they're accessed. So reading `ForageSDK.shared.merchantId` will run `ForageSDK.config?.merchantId ?? ""` under the hood, ensuring it's always using the latest value. 

It also makes these properties readonly since they aren't actually storing a value themselves — 
![CleanShot 2024-08-15 at 12 13 22@2x](https://github.com/user-attachments/assets/3f9acb14-a7d2-43ea-8897-e40f6d9a1eb8)